File: packages/common_native_channel/macos/Classes/CommonFeaturesInterface.swift
//
//  CommonFeaturesInterface.swift
//  common_native_channel
//
//  Created by admin on 2026/4/3.
//

import Foundation
import FlutterMacOS

protocol CommonFeaturesInterface: NSObject {
    var methods: [String] { get }

    var channel: FlutterMethodChannel { get }
    
    init(channel:FlutterMethodChannel)
    
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult)
    
}

File: packages/common_native_channel/android/src/main/kotlin/com/godcount/common_native_channel/features/CommonFeaturesInterface.kt
package com.godcount.common_native_channel.features

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

interface CommonFeaturesInterface {
    val methods: Array<String>
    val channel: MethodChannel
    fun handle(call: MethodCall, result: MethodChannel.Result)
}